api4/group.go

```diff
@@ -0,0 +1,450 @@
+	}
+
+	groupSyncables, err := c.App.GetGroupSyncables(c.Params.GroupId, syncableType)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if len(groupSyncables) == 0 {
+		w.Write([]byte("[]"))
+		return
+	}
+
+	b, marshalErr := json.Marshal(groupSyncables)
+	if marshalErr != nil {
+		c.Err = model.NewAppError("Api4.getGroupSyncables", "api.group.marshal_error", nil, marshalErr.Error(),
+			http.StatusNotImplemented)
+		return
+	}
+
+	w.Write(b)
+}
+
+func patchGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireGroupId()
+	if c.Err != nil {
+		return
+	}
+
+	c.RequireSyncableId()
+	if c.Err != nil {
+		return
+	}
+	syncableID := c.Params.SyncableId
+
+	c.RequireSyncableType()
+	if c.Err != nil {
+		return
+	}
+	syncableType := c.Params.SyncableType
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.patchGroupSyncable", "api.group.io_error", nil, err.Error(),
+			http.StatusNotImplemented)
+	}
+
+	var patch *model.GroupSyncablePatch
+	err = json.Unmarshal(body, &patch)
+	if err != nil || patch == nil {
+		c.SetInvalidParam(fmt.Sprintf("Group[%s]Patch", syncableType.String()))
+		return
+	}
+
+	if c.App.License() == nil || !*c.App.License().Features.LDAPGroups {
+		c.Err = model.NewAppError("Api4.patchGroupSyncable", "api.group.license.error", nil, "",
+			http.StatusNotImplemented)
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	var appErr *model.AppError
+	groupSyncable, appErr := c.App.GetGroupSyncable(c.Params.GroupId, syncableID, syncableType)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	groupSyncable.Patch(patch)
+
+	groupSyncable, appErr = c.App.UpdateGroupSyncable(groupSyncable)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	b, marshalErr := json.Marshal(groupSyncable)
+	if marshalErr != nil {
+		c.Err = model.NewAppError("Api4.patchGroupSyncable", "api.group.marshal_error", nil, marshalErr.Error(),
+			http.StatusNotImplemented)
+		return
+	}
+
+	w.Write(b)
+}
+
+func unlinkGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireGroupId()
+	if c.Err != nil {
+		return
+	}
+
+	c.RequireSyncableId()
+	if c.Err != nil {
+		return
+	}
+	syncableID := c.Params.SyncableId
+
+	c.RequireSyncableType()
+	if c.Err != nil {
+		return
+	}
+	syncableType := c.Params.SyncableType
+
+	if c.App.License() == nil || !*c.App.License().Features.LDAPGroups {
+		c.Err = model.NewAppError("Api4.unlinkGroupSyncable", "api.group.license.error", nil, "",
+			http.StatusNotImplemented)
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	_, err := c.App.DeleteGroupSyncable(c.Params.GroupId, syncableID, syncableType)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+	w.Write(nil)
```

Comment:
Don't we normally use `ReturnStatusOK` in these kinds of situation?

Reply:
I thought no content was more apt but if it's a matter of consistency am cool with changing it.

Reply:
Yeah, I think it's worth changing for consistency's sake.